reyna/src/com/b2msolutions/reyna/Dispatcher.java

```diff
@@ -49,20 +58,34 @@ protected Result sendMessage(Message message, HttpPost httpPost, HttpClient http
         return this.tryToExecute(httpPost, httpClient);
     }
 
-    public static Result canSend(Context context) {
+    public static Result canSend(Context context) throws ParseException {
+        // *****************************
+        // TODO all the checks here should be combined with our window blackout feature
+        // *****************************
+
         ConnectivityManager connectivityManager = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
         NetworkInfo info = connectivityManager.getActiveNetworkInfo();
-        if(info == null || !info.isConnectedOrConnecting()) {
+        if (info == null || !info.isConnectedOrConnecting()) {
             return Result.NOTCONNECTED;
         }
+        int type = info.getType();
+        Preferences preferences = new Preferences(context);
+
+        BlackoutTime blackoutTime = new BlackoutTime(context);
+        if (canSendOnWlanCharging(context, type, preferences, blackoutTime)) return Result.BLACKOUT;
+        if (canSendOnWlanDischarging(context, type, preferences, blackoutTime)) return Result.BLACKOUT;
```

Comment:
Before a refactor it looked like you are proposing, in our opinion it was less readable than the current solution.